geoportal/c2cgeoportal_geoportal/lib/caching.py

```diff
@@ -187,6 +187,10 @@ class Cache(Enum):
     PRIVATE = 2
     # See PUBLIC_NO and PRIVATE
     PRIVATE_NO = 3
+    # Sane as PUBLIC but if we are authenticated we are private
+    PUBLIC_CAN_PRIVATE = 4
```

Comment:
IMHO
```suggestion
    PUBLIC_OR_PRIVATE = 4
```

Reply:
For me, the deepening is
PUBLIC but the same URL can be private
=> CAN, not OR...
@arnaud-morvan does is looks good to you?

Reply:
Here we use a qualifier of the reponse content.

so it **can** be PUBLIC **OR** PRIVATE, depending on user authentication.

public can private => private is not verb, so it is not grammatically correct

The response content is "PUBLIC", "PRIVATE", or "PUBLIC_OR_PRIVATE", but it cannot be "PUBLIC_CAN_PRIVATE".

Reply:
or it can be PRIVATE_IF_AUTHENTICATED or PUBLIC_IF_ANONYMOUS

Reply:
The same view can return 
`PUBLIC_CAN_PRIVATE` or `PRIVATE` (depending on user authentication decided by the view).
Then `PUBLIC_OR_PRIVATE` is `PUBLIC` (except for the `Vary`) then it not a good name...
What about `PUBLIC_CAN_BE_PRIVATE`?

Reply:
In fact I'm not sure about why we need this, in case PRIVATE => if user is not authenticated, why not returning same as PUBLIC ?

PUBLIC => response content does not depend on authentication, even if user is authenticated, we can put it in a public cache.

PRIVATE => response depends on authentication, if user is authenticated, does not put it in a public cache.

Reply:
The PUBLIC/PRIVATE is used to manage the public caches (in Proxy)...

Reply:
in case we are in intranet (for example for roles) we can consider as authenticated